### PR TITLE
chore: Minor Refactoring

### PR DESCRIPTION
**Pull request by rish691**
Move out evaluation code from MetricAnomalyDetector to AlertRuleEvaluator

### CODE REVIEW

File: metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/AlertRuleEvaluatorTest.java [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+                          {
+                            "apiId1", "/login",
+                          },
+                          {"apiId2", "/checkout"}
+                        }))
+                .iterator());
+
+    AlertRuleEvaluator alertRuleEvaluator =
+        new AlertRuleEvaluator(config, attributesServiceClient, queryServiceClient);
+    //    Assertions.assertTrue(metricAnomalyDetector.process(alertTaskBuilder.build()));
```
Comment (sarthak77) [inline metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/AlertRuleEvaluatorTest.java:192]: Will this be done in a separate PR ?

File: metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetectorTest.java [hunk 2]
```diff
@@ -1,20 +1,14 @@
 package org.hypertrace.alert.engine.metric.anomaly.detector;
 
-import static org.mockito.ArgumentMatchers.eq;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.when;
+import static org.hypertrace.alert.engine.metric.anomaly.detector.AlertRuleEvaluatorTest.createLeafFilter;
+import static org.hypertrace.alert.engine.metric.anomaly.detector.AlertRuleEvaluatorTest.createLhsExpression;
```
Comment (sarthak77) [inline metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetectorTest.java:4]: Can/should we create a separate utility class for these functions ?

Comment (rish691) [reply metric-anomaly-detector/src/test/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetectorTest.java:4]: We do it at a later point

**Review by sarthak77 — APPROVED** [review]

**Review by kotharironak — COMMENTED** [review]
lgtm
